echopop/extensions/diagnostics.py

```diff
@@ -0,0 +1,510 @@
+
+    def mesh_cropping_results(self) -> pn.Column:
+
+        # Get the datasets
+        DATAINPUT = {
+            "Latitude/longitude": {
+                "cropped": (
+                    self.data.analysis["kriging"]["mesh_df"].copy()
+                    if "kriging" in self.data.analysis
+                    else {}
+                ),
+                "full": self.data.input["statistics"]["kriging"]["mesh_df"].copy(),
+            },
+            "Isobath-referenced": {
+                "cropped": self.data.analysis["kriging"]["mesh_df"].copy(),
+            },
+        }
+
+        # Get the isobath data
+        isobath_df = self.data.input["statistics"]["kriging"]["isobath_200m_df"]
+
+        # Amend the column names, if needed
+        [
+            v.rename(
+                columns={
+                    col: (
+                        "longitude"
+                        if "longitude" in col
+                        else "latitude" if "latitude" in col else col
+                    )
+                    for col in v.columns
+                },
+                inplace=True,
+            )
+            for k, v in DATAINPUT["Latitude/longitude"].items()
+        ]
+
+        # Initialize widget options
+        # ---- Variable options
+        var_options = ["Latitude/longitude", "Isobath-referenced"]
+        # ---- Get column names
+        var_columns = {
+            "Latitude/longitude": ["longitude", "latitude"],
+            "Isobath-referenced": ["x", "y"],
+        }
+        # ---- Tile options
+        var_tiles = {
+            "Latitude/longitude": self.tile,
+            "Isobath-referenced": None,
+        }
+        # ---- Axis options
+        var_axes = {
+            "Latitude/longitude": {
+                "xlab": "Longitude (\u00B0E)",
+                "ylab": "Latitude (\u00B0N)",
+            },
+            "Isobath-referenced": {
+                "xlab": "Normalized x",
+                "ylab": "Normalized y",
+            },
+        }
+
+        # Initialize `panel` extension
+        pn.extension()
+
+        # Create dropdown selector
+        var_dropdown = pn.widgets.Select(name="Variable", options=var_options)
+
+        # Create plotting function
+        @pn.depends(selection=var_dropdown)
+        def update_plot(selection):
+
+            # Parse dataset
+            data = DATAINPUT[selection]
+            # ---- Get tile source
+            tile = var_tiles[selection]
+            # ----  Get x-axis label
+            xlab = var_axes[selection]["xlab"]
+            # ---- Get y-axis label
+            ylab = var_axes[selection]["ylab"]
+            # ---- Get coordinates
+            coord = var_columns[selection]
+
+            # Initialize plotting layers
+            plot_layers = []
+
+            # Plot the data
+            # ---- Base tile layer
+            if tile:
+                plot_layers.append(tile)
+            # ---- Full mesh
+            if "full" in data:
```

Comment:
Just adding this for readability
```suggestion
            if "full" in data:  # not shown for isobath-referenced coord
```